### PR TITLE
Slu/agreement

### PR DESCRIPTION
**Pull request by sluFicodes**
Agreement API implemented with all of its dependency objects

### CODE REVIEW

File: agreement/pom.xml [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.fiware.tmforum</groupId>
+            <artifactId>common</artifactId>
+            <version>${project.version}</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.fiware.tmforum</groupId>
+            <artifactId>mapping</artifactId>
```
Comment (wistefan) [inline agreement/pom.xml:31]: Hi, I'm sorry, but your basically one day to late:) This dependency(even thought its still the same code), was extracted into a library. Please update to this.

Comment (sluFicodes) [reply agreement/pom.xml:31]: Good morning, I have updated the mapping libraries from the agreement module

**Comment by sluFicodes** [comment]
mapping libraries updated but after updating the new root pom.xml, mvn install -Pdev gives me an error because apparently, maven tried to deploy the same container again for each module: after party catalog building success the following module, product inventory, failed with this message: "Failed to execute goal io.kokuwa.maven:k3s-maven-plugin:1.0.0:run (prepare-test-cluster) on project product-inventory: Container with id '491122dc24c4' found. Please remove that container or set 'k3s.failIfExists' to false". The previous version of kokuwa (0.7.0) can detect through the creation goal (1.0.0 doesn't have it) if a container with the same id has been created and skip that goal, you can notice that via this log: "[INFO] Container with id '5667b153e5eb6b2fe7364d261ab7db9a6857ca9319772f52ad3a91b9e247a0ec' found, skip creating".

**Comment by wistefan** [comment]
Hi, that should be fixed now. The tests succeed, they are only red since the token for reporting the coverage is not available on your fork.

**Review by wistefan — APPROVED** [review]

**Pull request merged**